File: src/main/java/io/codeka/gaia/registries/github/GitHubRawContent.java
package io.codeka.gaia.registries.github;

import io.codeka.gaia.registries.RegistryRawContent;

import java.util.regex.Pattern;

public class GitHubRawContent extends RegistryRawContent {
    @Override
    protected final Pattern getPattern() {
        return Pattern.compile("^http[s]?://[www.]?github.com(.*).git$");
    }

    @Override
    protected String getTemplateRawUrl() {
        return "https://raw.githubusercontent.com{0}/{1}";
    }
}

File: src/main/java/io/codeka/gaia/registries/github/GitHubRawContent.kt
package io.codeka.gaia.registries.github

import io.codeka.gaia.registries.RegistryRawContent
import io.codeka.gaia.registries.RegistryType
import org.springframework.stereotype.Component
import org.springframework.web.client.RestTemplate

@Component
class GitHubRawContent(restTemplate: RestTemplate) : RegistryRawContent(RegistryType.GITHUB, restTemplate)
